webviz_subsurface/plugins/_co2_leakage/_utilities/co2volume.py

```diff
@@ -0,0 +1,136 @@
+import itertools
+import pathlib
+from enum import Enum
+from typing import Dict
+
+import pandas
+import numpy as np
+import plotly.express as px
+import plotly.graph_objects as go
+
+
+class _Columns(Enum):
+    REALIZATION = "realization"
+    VOLUME = "volume"
+    CONTAINMENT = "containment"
+    VOLUME_OUTSIDE = "volume_outside"
+
+
+def _read_dataframe(realization_path: str, relpath: str):
+    return pandas.read_csv(pathlib.Path(realization_path) / relpath)
+
+
+def _read_terminal_co2_volumes(realization_paths: Dict[str, str], table_relpath: str):
+    records = []
+    for rz_name, rz_path in realization_paths.items():
+        try:
+            df = _read_dataframe(rz_path, table_relpath)
+        except FileNotFoundError:
+            continue
+        last = df.iloc[np.argmax(df["date"])]
+        label = str(rz_name)
+        records += [
+            (label, last["co2_inside"], "inside", 0.0),
+            (label, last["co2_outside"], "outside", last["co2_outside"]),
+        ]
+    df = pandas.DataFrame.from_records(
+        records,
+        columns=[
+            _Columns.REALIZATION.value,
+            _Columns.VOLUME.value,
+            _Columns.CONTAINMENT.value,
+            _Columns.VOLUME_OUTSIDE.value,
+        ]
+    )
+    df.sort_values(_Columns.VOLUME_OUTSIDE.value, inplace=True, ascending=True)
+    return df
+
+
+def _read_co2_volumes(realization_paths: Dict[str, str], table_relpath: str):
+    return pandas.concat([
+        _read_dataframe(rz_path, table_relpath).assign(realization=rz_name)
+        for rz_name, rz_path in realization_paths.items()
+    ])
+
+
+def generate_co2_volume_figure(
+    realization_paths: Dict[str, str],
+    height: int,
+    width: int,
+    table_relpath: str,
+):
+    df = _read_terminal_co2_volumes(realization_paths, table_relpath)
+    fig = px.bar(
+        df,
+        y=_Columns.REALIZATION.value,
+        x=_Columns.VOLUME.value,
+        color=_Columns.CONTAINMENT.value,
+        title="End-state CO2 containment",
+        orientation="h",
+        category_orders={_Columns.CONTAINMENT.value: ["outside", "inside"]},
+        color_discrete_sequence=["#dd4300", "#006ddd"],
+    )
+    fig.layout.height = height
```

Comment:
Not sure if it is intended, but these size attributes generates somewhat small figures.
![image](https://user-images.githubusercontent.com/16436291/189523727-9b973476-8ca7-402a-812e-53b6f229902a.png)

Without:
![image](https://user-images.githubusercontent.com/16436291/189523915-1a6c62a3-fb5d-47f8-a3b1-98d248e8aab0.png)

You can also consider replacing `dcc.Graph` with the `wcc.Graph` wrapper which will add responsive resizing.
https://github.com/equinor/webviz-core-components/blob/master/webviz_core_components/wrapped_components/graph.py

Reply:
Removed fixed height/width and changed to wcc.Graph as suggested. Looks better now!